src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -122,7 +122,7 @@ private PerAnalyzerState GetAnalyzerState(DiagnosticAnalyzer analyzer)
                     OnCompilationEventsGenerated_NoLock(getCompilationEvents(eventQueue, additionalFiles));
                 }
             }
-            catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e))
+            catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, cancellationToken))
```

Comment:
📝 All uses of `ReportAndPropagateUnlessCanceled` are safe to change since it can only impact telemetry.